Moq.AutoMock/MockExtensions.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Moq.Language.Flow;
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Moq.AutoMock
+{
+    /// <summary>
+    /// TODO
+    /// </summary>
+    public static class MockExtensions
+    {
+        /// <summary>
+        /// Specifies a setup on the mocked type for a call to a void method. 
+        /// All parameters are filled with <see cref ="It.IsAny" /> according to the parameter's type.
+        /// </summary>
+        /// <remarks>
+        /// This may only be used on methods that are not overloaded.
+        /// </remarks>
+        /// <typeparam name="T">Type of the mock</typeparam>
+        /// <param name="mock">The mock</param>
+        /// <param name="methodName">The name of the expected method invocation.</param>
+        /// <exception cref="ArgumentNullException">When mock or methodName is null.</exception>
+        /// <exception cref="MissingMethodException">Thrown when no method with methodName is found.</exception>
+        /// <exception cref="AmbiguousMatchException">Thrown when more that one method matches the passed method name.</exception>
+        /// <returns></returns>
+        public static ISetup<T> SetupWithAny<T>(this Mock<T> mock, string methodName)
+            where T : class
+        {
+            if (mock is null)
+            {
+                throw new ArgumentNullException(nameof(mock));
+            }
+
+            if (methodName is null)
+            {
+                throw new ArgumentNullException(nameof(methodName));
+            }
+
+            LambdaExpression lambdaExpression = GetExpression<T>(methodName);
+
+            MethodInfo setupMethod = mock.GetType().GetMethods()
+                .Single(x => x.Name == nameof(Mock<object>.Setup) && x.ReturnType.GetGenericArguments().Length == 1);
```

Comment:
The user could benefit from more clear messaging here. We don't support overloads. Might as well explain why we don't, it certainly wouldn't hurt.

Reply:
I agree, I think this comment was meant to be further down where the `Type.GetMethod` throws an `AmbiguousMatchException`. Updated that with a more accurate message to indicate that we don't support overloads.